pandas/tests/io/parser/compression.py

```diff
@@ -166,6 +166,21 @@ def test_read_csv_infer_compression(self):
 
         inputs[3].close()
 
+    def test_read_csv_infer_compression_pathlib(self):
+        """
+        Test that compression is inferred from pathlib.Path paths.
+        """
+        try:
+            import pathlib
+        except ImportError:
+            pytest.skip('need pathlib to run')
+        expected = self.read_csv(self.csv1, index_col=0, parse_dates=True)
+        for extension in '', '.gz', '.bz2':
+            path = pathlib.Path(self.csv1 + extension)
+            df = self.read_csv(
+                path, index_col=0, parse_dates=True, compression='infer')
```

Comment:
@jreback, Just to make sure we're on the same page. All path types would be:

+ local paths via strings
+ local paths via __fspath__ defining objects
+ URLs

All compression types would be None, xz, bz2, zip, and gzip. But in all cases, we'd be inferring compression?

Just want to make sure the scope of this test should be primarily `compression='infer'`? Any more guidance will be appreciated.